src/main/java/net/sf/jabref/gui/prrv/ReferenceRelationship.java

```diff
@@ -0,0 +1,136 @@
+import prefux.data.Graph;
+
+/**
+ * @author Daniel Brühl
+ */
+public class ReferenceRelationship {
+
+    // List to store the temporary data for graph visualisation
+    private List<BibEntry> pureEntryList = JabRefGUI.getMainFrame().getCurrentBasePanel().getBibDatabaseContext().getDatabase().getEntries();
+    public List<prefux.data.Node> entryNodeList = new ArrayList<>();
+    public List<prefux.data.Node> externalNodeList = new ArrayList<>();
+    private List<prefux.data.Edge> currentEdgeList = new ArrayList<>();
+
+    /**
+     * Generate necessary node for prefux graph reader.
+     * For every BibTex entry and reference to external papers generate one node
+     */
+    public void parseBibTexForReferences(Graph g) {
+
+        if (JabRefGUI.getMainFrame().getCurrentBasePanel().getBibDatabaseContext().getDatabase().getEntries().isEmpty()) {
+            FXDialogs.showInformationDialogAndWait(Localization.lang("No_references_found"), "Please load a database");
+        } else {
+            // Init - Create nodes for bibtex entries
+            for (int sourceID = 0; pureEntryList.size() > sourceID; sourceID++) {
+                if (pureEntryList.get(sourceID).getField("cites").isPresent() && pureEntryList.get(sourceID).getField("bibtexkey").isPresent()) {
+                    // Split cites for citeCount
+                    String[] citeLines = pureEntryList.get(sourceID).getField("cites").get().split(",");
+                    // Create node of bibtex entry
+                    System.out.println(pureEntryList.get(sourceID).getField("bibtexkey").get());
+                    entryNodeList.add(createNode(sourceID, pureEntryList.get(sourceID).getField("bibtexkey").get(), citeLines.length + 1, "I", g));
+                }
+            }
+
+            // Create edges & external nodes
+            for (int sourceID = 0; pureEntryList.size() > sourceID; sourceID++) {
+                if (pureEntryList.get(sourceID).getField("cites").isPresent()) {
+                    // Split cites
+                    String[] citeLines = pureEntryList.get(sourceID).getField("cites").get().split(",");
+
+                    // Go trough all cites lines of your current entry and check if it refers to some of your bibtex entries
+                    for (String citeLine : citeLines) {
+
+                        // boolean to know if cites was found
+                        boolean foundCite = false;
+
+                        // Skip empty lines
+                        if (citeLine != null) {
+                            // Check every other entry
+                            for (int targetID = 0; pureEntryList.size() > targetID; targetID++) {
+                                // Don't refer to yourself
+                                // Cite matching with bibtexkey of any bibtex entry?
+                                if (citeLine.trim().equals((pureEntryList.get(targetID).getField("bibtexkey").get().trim())) && sourceID != targetID) {
```

Comment:
As far as I know the `getField()` method of an entry returns an optional which can also be null. Can it happen that at his stage an entry in the pureEntryList has no BibTeX-key?